src/main/java/com/aws/greengrass/mqttclient/MqttClient.java

```diff
@@ -422,9 +425,11 @@ private void triggerSpooler() {
      * @throws ExecutionException   if an error occurs
      * @throws InterruptedException if the thread is interrupted while unsubscribing
      * @throws TimeoutException     if the request times out
+     * @throws MqttRequestException if request topic is not valid
      */
     public synchronized void unsubscribe(UnsubscribeRequest request)
-            throws ExecutionException, InterruptedException, TimeoutException {
+            throws ExecutionException, InterruptedException, TimeoutException, MqttRequestException {
```

Comment:
You must not change the signature of these methods. These are publicly consumed already by plugins (lambda). Wrap it as an ExecutionException.